common-service-core/src/main/java/com/ca/mfaas/message/core/AbstractMessageService.java

```diff
@@ -0,0 +1,136 @@
+            return Message.of(key, messageTemplate, parameters);
+        }
+    }
+
+    /**
+     * Creates list {@link Message} with list of {@link Message}.
+     *
+     * @param key        of message in messages.yml file
+     * @param parameters list that contains arrays of parameters
+     * @return {@link Message}
+     */
+    @Override
+    public List<Message> createMessage(String key, List<Object[]> parameters) {
+        return parameters.stream()
+            .filter(Objects::nonNull)
+            .map(ob -> createMessage(key, ob))
+            .collect(Collectors.toList());
+    }
+
+    /**
+     * Validate and add a {@link MessageTemplates} to the {@link MessageTemplateStorage}.
+     * @param messageTemplates the list of message templates
+     */
+    protected final void addMessageTemplates(MessageTemplates messageTemplates) {
+        validateMessageTemplates(messageTemplates);
+        messageTemplateStorage.addMessageTemplates(messageTemplates);
+    }
+
+    /**
+     * Validate {@link MessageTemplates} by checking that there are not occurrences of {@link MessageTemplate} with the same key.
+     * If there are, an exception is thrown.
+     * @param messageTemplates the list of message templates
+     * @throws DuplicateMessageException
+     */
+    private void validateMessageTemplates(MessageTemplates messageTemplates) {
+        String existedMesageTemplates = messageTemplates.getMessages()
+            .stream()
+            .collect(
+                Collectors.groupingBy(
+                    MessageTemplate::getNumber,
+                    Collectors.counting()
+                )
+            )
+            .entrySet()
+            .stream()
+            .filter(entry -> entry.getValue() > 1)
+            .map(Map.Entry::getKey)
+            .collect(Collectors.joining(","));
+
+        if (!existedMesageTemplates.equals("")) {
+            String exceptionMessage = String.format("Message template with number [%s] already exists", existedMesageTemplates);
+            throw new DuplicateMessageException(exceptionMessage);
+        }
+    }
+
+    /**
+     * Validate {@link MessageTemplate} by checking if the message with the a specific key exists in the {@link MessageTemplateStorage}.
+     * If it does not, the method returns an invalid key message.
+     * @param key the message key
+     * @return {@link MessageTemplate}
+     */
+    private MessageTemplate validateMessageTemplate(String key) {
+        return messageTemplateStorage.getMessageTemplate(key)
+            .orElseGet(() -> {
+                log.debug("Invalid message key '{}' was used. Please resolve this problem.", key);
+                return messageTemplateStorage.getMessageTemplate(Message.INVALID_KEY_MESSAGE).orElseGet(
+                    this::getInvalidMessageTemplate);
+            });
+    }
+
+    /**
+     * Return the invalid key message {@link MessageTemplate}.
```

Comment:
Is that the imperative mood, or erroneously written the third-person singular of 'to return'?

Reply:
Right, it was not meant to be imperative, sorry :D